### DISCUSSION TITLE
Heatmaps in Home Assistant possible?

### DISCUSSION THREAD

**Discussion by JulianPSLearner** [RomRider/apexcharts-card#748] [Q&A]
Hi guys, I was looking for hours for examples for the heatmap. Are there any? I found on the apex chart side few examples but not customized for HA.

I wanted a heatmap that shows me on the X-axis day 1-30 and on y-axis I want to see the different room temperatur sensors.
I want to see three color variations depending on the heat.

This is what I got so far. Unfortunately no heatmap shows up at all :(
Any help would be appreciated:

> type: custom:apexcharts-card
apex_config:
  chart:
    type: heatmap
  plotOptions:
    heatmap:
      colorScale:
        ranges:
          - from: -50
            to: 20
            color: '#00A100'
            name: 'Below 20°C'
          - from: 20
            to: 23
            color: '#128FD9'
            name: '20°C to 23°C'
          - from: 23
            to: 50
            color: '#FF0000'
            name: 'Above 23°C'
  yaxis:
    categories:
      - 'January'
      - 'February'
      - 'March'
      - 'April'
      - 'May'
      - 'June'
      - 'July'
      - 'August'
      - 'September'
      - 'October'
      - 'November'
      - 'December'
  xaxis:
    categories: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30]
header:
  show: true
  title: Temperature Heatmap
  show_states: true
  colorize_states: true
series:
  - entity: sensor.og_kind_3_thermo_0467_1_actual_temperature
    name: 'Temperature'
    data_generator: |
      function() {
        const data = [];
        const months = 12;
        const days = 30;
        const sensorData = entity.attributes['history'][0]; // Example for historical data

        for (let month = 1; month <= months; month++) {
          for (let day = 1; day <= days; day++) {
            const temp = sensorData.find(record => record.month === month && record.day === day);
            data.push({
              x: day,
              y: month,
              value: temp ? temp.temperature : null // Replace with actual data fetching logic
            });
          }
        }
        return data;
      }

**Comment by TheSerapher** [comment]
I tried heat maps myself but I don't think they are properly supported by this card. Went with `plotly` instead to have them. Would be great to have them natively in this card but I don't think that's happening soon.
